geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/MemberIdentifierImpl.java

```diff
@@ -0,0 +1,978 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.distributed.internal.membership.api;
```

Comment:
gawd this (class) really goes into the membership *API*. That makes everything this class _does_ part of the membership contract right?

class inheritance coming home to roost. I guess the alternative is to put a _factory_ in the API and hide this impl in the membership module implementation package. Then `InternalDistributedMember` would _have_ a `MemberIdentifierImpl` through a `MemberIdentifier`reference.

Problem with that idea is that `MemberIdentifierImpl` has a ton of methods not on `MemberIdentifier`. That implies we'd need another, wider, interface to implement _those_ methods (maybe a sub-interface of `MemberIdentifier`) ugh.

sorry for the stream of consciousness feedback. I'll try to hone it and get back to you…